### PR TITLE
Add ssl option to app channel

### LINKED ISSUE #1883: failed to open grpc channel to app for ASP.NET Core gRPC service
## In what area(s)?

/area runtime
> /area operator
> /area placement
> /area docs
> /area test-and-release

## What version of Dapr?

version 0.9.0 commit 6babe85-dirty

## Expected Behavior

Dapr runtime should connect app which be developed by ASP.NET Core gRPC service

## Actual Behavior

Got error: level=warning msg="failed to open grpc channel to app: error establishing connection to app grpc on port 5001: context deadline exceeded"

And client report "app channel is not initialized"

## Steps to Reproduce the Problem

I added a grpc sample into dotnet-sdk in here (https://github.com/heavenwing/dapr-dotnet-sdk/tree/grpcsample)

My sample project is "GrpcServiceSample" which use ASP.NET Core gRPC service to host grpc server.

The GrpcClient can connect this grpc sample (run with "dotnet run"), and invoke method.

But when I run GrpcServiceSample in Dapr with this command "dapr run --app-id grpcsample --app-port 5001 --protocol grpc -- dotnet run", I got above error.

And I modified DaprClient to invoke my grpc sample, it also reports error when run in Dapr with this command "dapr run --app-id client -- dotnet run"

whole log info in this log file
[dapr grpc sample.log](https://github.com/dapr/dapr/files/5049636/dapr.grpc.sample.log)

### PR DESCRIPTION
**Pull request by yaron2**
This PR adds the option for a user to specify whether their application (HTTP or gRPC enabled) is listening on an SSL port.

If SSL mode is enabled, Dapr will use an "https://" prefix for the app channel and attempt an SSL connection.

Closes #1883

### CODE REVIEW

File: pkg/grpc/grpc.go [hunk 1]
```diff
@@ -112,6 +112,15 @@ func (g *Manager) GetGRPCConnection(address, id string, namespace string, skipTL
 	defer cancel()
 
 	dialPrefix := GetDialAddressPrefix(g.mode)
+	if sslEnabled {
+		dialPrefix = "https://"
```
Comment (mukundansundar) [inline pkg/grpc/grpc.go:116]: for Kubernetes also does it need to change to https?

File: pkg/runtime/config.go [hunk 2]
```diff
@@ -14,6 +14,9 @@ import (
 // Protocol is a communications protocol
 type Protocol string
 
+// Scheme is a URI address scheme
+type Scheme string
```
Comment (mukundansundar) [inline pkg/runtime/config.go:18]: unused?

File: pkg/grpc/grpc.go (hunk 1 above)
Comment (youngbupark) [inline pkg/grpc/grpc.go:116]: No we do not need `https://` scheme prefix. https:// is not eligible scheme for grpc client.
Even if Kestrel server (HTTP/2) is used as gRPC transport layer, gRPC client handles gRPC protocol.

File: pkg/runtime/config.go [hunk 3]
```diff
@@ -29,6 +32,7 @@ const (
 	DefaultMetricsPort = 9090
 	// DefaultAllowedOrigins is the default origins allowed for the Dapr HTTP servers
 	DefaultAllowedOrigins = "*"
+	// DefaultAppScheme is the default URI scheme when Dapr calls the application
```
Comment (youngbupark) [inline pkg/runtime/config.go:35]: where is `DefaultAppScheme` ? :)

**Review by youngbupark — CHANGES_REQUESTED** [review]

File: pkg/grpc/grpc.go (hunk 1 above)
Comment (yaron2) [reply pkg/grpc/grpc.go:116]: Yes, this is going to the app, not to another Dapr instance.

File: pkg/runtime/config.go (hunk 3 above)
Comment (yaron2) [reply pkg/runtime/config.go]: A remain from an older implementation, removed now.

File: pkg/grpc/grpc.go (hunk 1 above)
Comment (yaron2) [reply pkg/grpc/grpc.go]: Changed.

File: pkg/runtime/config.go (hunk 2 above)
Comment (yaron2) [reply pkg/runtime/config.go]: A remain from an older implementation, removed now.

**Review by youngbupark — APPROVED** [review]

**Pull request merged**